### PR TITLE
Use `Filename.get_temp_dir_name` instead of `/tmp` for Windows compatibility

### PR DESCRIPTION
**Pull request by nojb**
We are thinking of switching to `ocamlformat` at LexiFi and have been experimenting with different potential workflows to handle the migration. In this context, I have been playing with this tool. However, since at LexiFi we don't use OPAM, big dependencies like `stdio` and `base` are inconvenient, so I did a small patch getting rid of this dependencies. If you are interested, you are welcome to integrate this!

Apart from the dependency issue there was a small portability bug: the use of `/tmp` on Windows is not correct, in this patch it is changed to `Filename.get_temp_dir_name()`.

Finally, note that the changes in this patch make the code 4.14-only (because they use the newly added `Out_channel`, `In_channel`, etc). If supporting old versions of OCaml is important for you, I think it should be doable to make a patch that supports old versions of OCaml (but I haven't tried it).

### CODE REVIEW

**Comment by nojb** [comment]
(Also: I only started testing the tool recently, so even if you are interested, you may want to hold off on merging the patch, in case other issues come up.)

**Comment by hhugo** [comment]
I'm happy to make users life easier. Ill wait a bit before merging in case there are other things to fix.

**Comment by hhugo** [comment]
I'm surprised base and stdio are a problem for you since there are also a dependency of ocamlformat

**Comment by hhugo** [comment]
You'll need to update the opam file.

File: src/setup_cmd.ml [hunk 1]
```diff
@@ -49,10 +49,12 @@ let merge =
       ; sprintf "git config --local merge.default 'mergefmt'"
       ]
     in
-    List.iter commands ~f:(fun line ->
+    List.iter
```
Comment (hhugo) [inline src/setup_cmd.ml]: Opening stdlabel would help reduce the diff

**Comment by nojb** [comment]
> I'm surprised base and stdio are a problem for you since there are also a dependency of ocamlformat

That's right, and eventually we may end up taking the step of integrating `base` and `stdio` into our build system. But for now, we vendor a __binary__ of `ocamlformat`. The binary is built using OPAM. However, our developers do not need to have OPAM (which we don't otherwise use) in their machines.

File: src/setup_cmd.ml (hunk 1 above)
Comment (nojb) [reply src/setup_cmd.ml]: I pushed a commit with this change, but the reduction is small.

**Comment by nojb** [comment]
> > I'm surprised base and stdio are a problem for you since there are also a dependency of ocamlformat
> 
> That's right, and eventually we may end up taking the step of integrating `base` and `stdio` into our build system. But for now, we vendor a **binary** of `ocamlformat`. The binary is built using OPAM. However, our developers do not need to have OPAM (which we don't otherwise use) in their machines.

I took a quick look; integrating `base` and `stdio` seems quite easy in fact (last time I tried there were some difficulties compiling on Windows, but I guess these were ironed out). So perhaps it is not worth modifying `merge-fmt` at all.

I modified the PR to only contain the bug fix related to `/tmp`. Thanks!

**Pull request merged**